code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm

```diff
@@ -177,17 +177,20 @@ Difficulty: Medium
 	hitsound = 'sound/weapons/sear.ogg'
 	var/storm_type = /datum/weather/ash_storm
 	var/storm_cooldown = 0
-	var/static/list/excluded_areas = list(/area/reebe/city_of_cogs)
+	var/static/list/allowed_areas = list(/area/lavaland/surface/outdoors)
 
 /obj/item/staff/storm/attack_self(mob/user)
 	if(storm_cooldown > world.time)
 		to_chat(user, "<span class='warning'>The staff is still recharging!</span>")
 		return
 
+	if(user.z != 5)
```

Comment:
dont hardcode this, use z level trait checks

Reply:
I'm not aware of us having those traits implemented. Every z-level check I've ever seen is a hardcoded check like this 🤔

Reply:
they are implemented. see maps.dm in the defines folder

Reply:
Followed your advice, looked up the define and could not figure out how to properly utilize it. Attempts to broke the staff, and this is again normal to be hardcoded here when checking an item user's current Z level. 

While I agree this would be better practice it's both beyond me and something in need of its own separate refactor touching everything else that is currently hardcoded as it is. `level_has_any_trait()` and `level_has_all_traits()` are both currently unused and I have no functional implementation of these to reference.